sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -326,6 +327,18 @@ public Mono<List<Address>> getServerAddressesViaGatewayAsync(
         String collectionRid,
         List<String> partitionKeyRangeIds,
         boolean forceRefresh) {
+
+        MetadataRequestRetryPolicy metadataRequestRetryPolicy = new MetadataRequestRetryPolicy(globalEndpointManager);
+        metadataRequestRetryPolicy.onBeforeSendRequest(request);
+
+        return BackoffRetryUtility.executeRetry(() -> this.getServerAddressesViaGatewayInternalAsync(
```

Comment:
same comments as above, I think we should combine with WebExceptionRetryPolicy

Reply:
Resolved in next iteration.